### PR TITLE
[Bug][SnowFlakeUtils] fix snowFlake bug

### PR DESCRIPTION
**Pull request by brave-lee**
fix getting same task code in bulk definition
#6658

### CODE REVIEW

**Comment by brave-lee** [comment]
@ruanwenjun @zhongjiajie @lenboo  Please confirm to get duplicate code

**Comment by zhongjiajie** [comment]
> @ruanwenjun @zhongjiajie @lenboo Please confirm to get duplicate code

Good job 👍  I would take a look right now

**Comment by zhongjiajie** [comment]
👍 I think it pass and no more duplicate code generate. I test locally by create 100 workflows and each workflow with 1000 tasks, all workflow created without any error anymore!

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/SnowFlakeUtilsTest.java [hunk 1]
```diff
@@ -23,7 +23,8 @@
     @Test
     public void testNextId() {
         try {
-            for (int i = 0; i < 5; i++) {
+            for (int i = 0; i < 100; i++) {
+                Thread.sleep(1);
```
Comment (zhongjiajie) [inline dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/SnowFlakeUtilsTest.java:27]: Is there any reason for us to use `sleep` here?
```suggestion
```

**Review by ruanwenjun — COMMENTED** [review]
LGTM.

It seems we don't need to calculate the `| machineId << MACHINE_LEFT` in line 73 every time?
I am not sure if java compiler will do this optimization, but it should be a constant?

**Comment by caishunfeng** [comment]
LGTM

File: dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/SnowFlakeUtilsTest.java (hunk 1 above)
Comment (brave-lee) [reply dolphinscheduler-common/src/test/java/org/apache/dolphinscheduler/common/utils/SnowFlakeUtilsTest.java:27]: This is a UT test to see if a duplicate is generated in the next millisecond

**Review by lenboo — APPROVED** [review]
+1

**Comment by brave-lee** [comment]
> LGTM.
> 
> It seems we don't need to calculate the `| machineId << MACHINE_LEFT` in line 73 every time? I am not sure if java compiler will do this optimization, but it should be a constant?

I think it should not be a constant, this is related to the current machine, `MACHINE_BIT = 2` can support four machines of api server

**Comment by ruanwenjun** [comment]
> > LGTM.
> > It seems we don't need to calculate the `| machineId << MACHINE_LEFT` in line 73 every time? I am not sure if java compiler will do this optimization, but it should be a constant?
> 
> I think it should not be a constant, this is related to the current machine, `MACHINE_BIT = 2` can support four machines of api server

Yes, I mean when we can calculate in the constructor, and then we can avoid calculate in each time execute nextId.
This is not a problem, this does not affect performance.

**Comment by brave-lee** [comment]
> > > LGTM.
> > > It seems we don't need to calculate the `| machineId << MACHINE_LEFT` in line 73 every time? I am not sure if java compiler will do this optimization, but it should be a constant?
> > 
> > 
> > I think it should not be a constant, this is related to the current machine, `MACHINE_BIT = 2` can support four machines of api server
> 
> Yes, I mean when we can calculate in the constructor, and then we can avoid calculate in each time execute nextId. This is not a problem, this does not affect performance.

You're right, can you contribute a new PR to optimize it ?

**Comment by zhongjiajie** [comment]
> fix getting same task code in bulk definition #6658

Hi @JinyLeeChina, I find out issue #6658 still open, maybe next time we should use keyword `closes #6658` or `close #6658` when we target to specific issue, it will auto close issue when PR is merged.
BTW, good fix 👍

**Comment by zhongjiajie** [comment]
And so sorry for raise discuss in https://github.com/apache/dolphinscheduler/pull/6744#discussion_r745250021 again, Technically, I do not think we should use `System.out.println` in our unit test. maybe we should declare a `set` to keep code generate before, and judge if new code wether in `set` or not.

**Comment by ruanwenjun** [comment]
> And so sorry for raise discuss in [#6744 (comment)](https://github.com/apache/dolphinscheduler/pull/6744#discussion_r745250021) again, Technically, I do not think we should use `System.out.println` in our unit test. maybe we should declare a `set` to keep code generate before, and judge if new code wether in `set` or not.

Yes, maybe we can set a check rule in checkstyle.xml to prevent use `System.out.println`

**Comment by zhongjiajie** [comment]
Could we avoid using `System.out.println` in test by checkstyle.xml? If so that would be great!

**Comment by ruanwenjun** [comment]
@zhongjiajie I think we can, in most cases we can use logger to replace the `System.out.println`.
A simple way is adding below check rule in checkstyle.xml
```
<module name="RegexpSingleline">
    <property name="format" value="System\..+\.println"/>
</module>
```

**Comment by kezhenxu94** [comment]
> Could we avoid using `System.out.println` in test by checkstyle.xml? If so that would be great!

@zhongjiajie take this as an example

https://github.com/apache/skywalking/blob/36d3fb1fe0fd6c6d04add6d91faf8c0f8890f2c4/apm-checkstyle/checkStyle.xml#L31-L34

**Pull request merged**